tracetypes/org.eclipse.tracecompass.incubator.opentracing.core/src/org/eclipse/tracecompass/incubator/opentracing/core/analysis/callstack/AsincCallStackStateProvider.java

```diff
@@ -0,0 +1,209 @@
+package org.eclipse.tracecompass.incubator.opentracing.core.analysis.callstack;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.tracecompass.incubator.callstack.core.instrumented.statesystem.InstrumentedCallStackAnalysis;
+import org.eclipse.tracecompass.incubator.internal.opentracing.core.event.IOpenTracingConstants;
+import org.eclipse.tracecompass.statesystem.core.ITmfStateSystemBuilder;
+import org.eclipse.tracecompass.statesystem.core.statevalue.TmfStateValue;
+import org.eclipse.tracecompass.tmf.core.event.ITmfEvent;
+import org.eclipse.tracecompass.tmf.core.event.ITmfEventField;
+import org.eclipse.tracecompass.tmf.core.statesystem.AbstractTmfStateProvider;
+import org.eclipse.tracecompass.tmf.core.statesystem.ITmfStateProvider;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+import org.eclipse.tracecompass.tmf.core.trace.TmfTraceUtils;
+
+/**
+ * Span Callstack state provider
+ *
+ * @author Fateme Faraji Daneshgar
+ *
+ */
+public class AsincCallStackStateProvider extends AbstractTmfStateProvider {
+
+    /**
+     * Thread attribute
+     *
+     * @since 2.0
+     */
+    public static final String PROCESSES = "Processes"; //$NON-NLS-1$
+
+    /**
+     * Unknown process ID
+     *
+     * @since 2.0
+     */
+    public static final int UNKNOWN_PID = -1;
+
+    /**
+     * Unknown name
+     *
+     * @since 2.0
+     */
+    public static final String UNKNOWN = "UNKNOWN"; //$NON-NLS-1$
+
+
+    private static final int MAX_STACK_DEPTH = 100000;
+
+    private final Map<String, Integer> fStackDepthMap;
+    private TreeMap<Long, List<Integer>> fPrevEvent;
```

Comment:
Ideally, we do not want to track any states outside of the state system as this makes it incompatible with certain backends.